sklearn/ensemble/_bagging.py

```diff
@@ -126,7 +123,11 @@ def _parallel_build_estimators(
         )
 
         # Draw samples, using sample weights, and then fit
+        support_sample_weight = has_fit_parameter(
+            ensemble.base_estimator_, "sample_weight"
+        )
         if support_sample_weight:
+            sample_weight = fit_params.get("sample_weight")
```

Comment:
I am wondering if we could maybe do a `pop`:

```python
try:
    sample_weight = fit_params.pop("sample_weight")
except KeyError:
    sample_weight = None
```

Then we don't need the `dict` comprehension below.

Reply:
I changed the code to use `fit_params.pop("sample_weight", None)`

Reply:
Did not know that `pop` as a "default" to be returned if not in the dict.